cmd/reports-controller/main.go

```diff
@@ -93,25 +106,47 @@ func createReportControllers(
 			))
 		}
 		if backgroundScan {
-			ctrls = append(ctrls, internal.NewController(
-				backgroundscancontroller.ControllerName,
-				backgroundscancontroller.NewController(
+			var backgroundScanController controllers.Controller
+			if validatingAdmissionPolicyReports {
+				backgroundScanController = backgroundscancontroller.NewController(
 					client,
 					kyvernoClient,
 					eng,
 					metadataFactory,
 					kyvernoV1.Policies(),
 					kyvernoV1.ClusterPolicies(),
+					kubeInformer.Admissionregistration().V1alpha1().ValidatingAdmissionPolicies(),
```

Comment:
Would it be a problem if the VAP feature gate is not enabled?

Reply:
Yes. if `--validatingAdmissionPolicyReports` is enabled and VAP feature gate isn't, then errors will exist.

Reply:
I'm checking if VAPs are registered in the API server or not.